src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -172,30 +172,141 @@ public class ChangeManagement implements Serializable {
+                                |        client: ${client}
+                                |        auth: basic
+                                |      credentials:
+                                |        username: env:ABAP_USER
+                                |        password: env:ABAP_PASSWORD
+                                |      app:
+                                |        name: ${applicationName}
+                                |        description: ${desc}
+                                |        package: ${abapPackage}
+                                |        transport: ${transportRequestId}
+                                |      exclude:
+                                |      - .*\\.test.js
+                                |      - internal.md
+                                |""" as CharSequence).stripMargin()
+
+        def deployConfigFile = 'ui5-deploy.yaml' // this is the default value assumed by the toolset anyhow.
+
+        // Revisit: should be located in a tmp dir inside '.pipeline'
+        script.writeFile file: deployConfigFile, text: deployConfig, encoding: 'UTF-8'
+
+        // 2.) create the call
+        // 2.1) prepare environment --> currently I assume a node default image. We need to start
+        //      as root, after that we can switch to a standard user (e.g. node/1000). Since we dont su
+        //      with '-l' flag the environment variables are presenved. This is important for the credentials.
+        //      Other approach would be to provide a derived image already containing the fiori upload deps.
+        //      With this the upload is faster, but we have to maintain the image.
+        // 2.2) the call in the narrower sense
+        //
+        // REVISIT: in case the customer uses a preconfigured image with the deploy tools already installed
+        //          there is no need for the npm call. --> in case the deployToolDependencies we should omit the
+        //          npm install call. In that case we should also drop the su and we don't launch the container as
+        //          root.
+
+        if (deployToolDependencies in List) {
+            deployToolDependencies = deployToolDependencies.join(' ')
+        }
 
-        if(rc != 0) {
-            throw new ChangeManagementException(
-                "Cannot upload file into transport request. Return code from  cm client: $rc.")
+        deployToolDependencies = deployToolDependencies.trim()
+
+        // --> Default config has been adjusted so that no dependencies needs to be installed.
+        // This is the case when an image is used which contains already all dependencies.
+        // In this case we don't invoke npm install and we run the image with the standard user
+        // already, since there is no need for being root. Hence we don't need to switch user also
+        // in the script.
+        boolean noInstall = deployToolDependencies.isEmpty()
+
+        def cmd = ['#!/bin/bash -e']
```

Comment:
If you define cmd as a List here, could you remove the `as Iterable`below?

Reply:
Does not work when defining a `List`. The deprecation warning would be the same. BUT: when we define it as  `Iterable` here, it works without deprecation warning. That is of course a more elegant approach ...

--> changes accordingly, thanks for raising the question!